tasks/register.yml

```diff
@@ -53,7 +53,10 @@
     --cache-s3-bucket-name "{{ gitlab_runner.cache_bucket_name }}"
     --cache-s3-insecure "{{ gitlab_runner.cache_insecure | default("false") }}"
     {% endif %}
-  when: gitlab_runner.name not in configured_runners.stderr
+  when: 
+    - gitlab_runner.name not in configured_runners.stderr
+    # Skip registration in test run when registration token is unset
+    - not ("molecule-notest" in ansible_skip_tags and gitlab_runner.registration_token | length == 0)
```

Comment:
Is it really necessary? See https://github.com/hifis-net/ansible-role-gitlab-runner/blob/8739f6bd1bcbcac4b53b2209ee72a71a0a981aa3/tasks/main.yml#L35

Reply:
This test checks if we're in a test run and if the registration token Is set. This way, we still run the registration on `main` and in our scheduled jobs. 
Do I get your point right?

Reply:
I thought we were not reaching this part of the config because when including `register.yml` the registration token length is checked before - see https://github.com/hifis-net/ansible-role-gitlab-runner/blob/8739f6bd1bcbcac4b53b2209ee72a71a0a981aa3/tasks/main.yml#L36
But maybe I'm wrong here...

Reply:
Good point, Thanks for the explanation.
Will check that!

Reply:
Indeed, this part was never reached. So the mentioned condition is not needed and thus removed.